File: easy/3622_check_divisibility_by_digit_sum_and_product/solution.cpp
class Solution {
public:
  bool checkDivisibility(int n) {
    int digitsSum = 0;
    int digitsProduct = 1;

    int nCopy = n;

    while (nCopy) {
      int digit = nCopy % 10;
      nCopy /= 10;

      digitsSum += digit;
      digitsProduct *= digit;
    }

    return n % (digitsSum + digitsProduct) == 0;
  }
};

File: easy/3622_check_divisibility_by_digit_sum_and_product/solution.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        digitsSum = 0
        digitsProduct = 1

        nCopy = n

        while nCopy:
            digit = nCopy % 10
            nCopy //= 10

            digitsSum += digit
            digitsProduct *= digit

        return n % (digitsSum + digitsProduct) == 0
